src/widget.cpp

```diff
@@ -1857,7 +1811,7 @@ void NWidgetVertical::AssignSizePosition(SizingType sizing, int x, int y, uint g
 
 		if (num_changing_childs == 0 && (flags & NC_BIGFIRST) && biggest_stepsize > 0) {
 			/* Second.5 loop: count how many children are of the updated biggest step size. */
-			for (NWidgetBase *child_wid = this->head; child_wid != nullptr; child_wid = child_wid->next) {
+			for (auto &child_wid : this->children) {
```

Comment:
feels like quite a lot of these for range loops could be const ?

Reply:
Hmm, well, actually they can ALL be const, but that looks weird...

Reply:
weird how? To me all range-for should be `const auto &`, anything else is "strange" and suggests that the object is being modified

Reply:
To me it looks like I shouldn't be able to change child_wid's variables nor call non-const functions. But I can, because it's the unique_ptr that's const, not the object that the unique_ptr points to.

That's why it looks odd.